js/main.js

```diff
@@ -1,2 +1,95 @@
+var photo = document.querySelector('#main-image');
+
+function updatePhoto(event) {
+  photo.setAttribute('src', event.target.value);
+  if (event.target.value === '') {
+    photo.setAttribute('src', 'images/placeholder-image-square.jpg');
+  }
+}
+
+urlInput.addEventListener('input', updatePhoto);
+
+var form = document.querySelector('#form');
+var title = document.querySelector('#title-text');
+var notes = document.querySelector('#notes-text');
+
+function submitListener(event) {
+  data.entries.unshift({
+    title: title.value,
+    photoURL: urlInput.value,
+    notes: notes.value,
+    entryID: data.nextEntryId
+  });
+  data.nextEntryId++;
+
+  entryFormContainer.className = 'container hidden';
+  entriesMainContainer.className = 'container';
+
+  photo.setAttribute('src', 'images/placeholder-image-square.jpg');
+  form.reset();
+}
+
+form.addEventListener('submit', submitListener);
+
+var entryUL = document.querySelector('#entry-list');
+var noEntriesMessage = document.querySelector('#no-entries');
+
+function entryAdd(entry) {
+  var entryListing = document.createElement('li');
+  var firstChild = entryUL.firstChild;
+  entryListing.className = 'row entry';
+  entryUL.insertBefore(entryListing, firstChild);
+
+  var entryImage = document.createElement('img');
+  entryImage.className = 'column-half';
+  entryImage.setAttribute('src', entry.photoURL);
+  entryListing.appendChild(entryImage);
+
+  var entryDiv = document.createElement('div');
+  entryDiv.className = 'column-half';
+  entryListing.appendChild(entryDiv);
+
+  var entryTitle = document.createElement('h2');
+  var entryTitleText = document.createTextNode(entry.title);
+  entryTitle.appendChild(entryTitleText);
+  entryDiv.appendChild(entryTitle);
+
+  var entryNotes = document.createElement('p');
+  var entryNotesText = document.createTextNode(entry.notes);
+  entryNotes.appendChild(entryNotesText);
+  entryDiv.appendChild(entryNotes);
+}
+
+window.addEventListener('DOMContentLoaded', domContentLoadedListener);
+
+function domContentLoadedListener(event) {
+  if (data.nextEntryId !== 1) {
+    noEntriesMessage.remove();
+  }
+  for (var i = (data.entries.length - 1); i >= 0; i--) {
+    entryAdd(data.entries[i]);
+  }
+}
+
+var entriesNavItem = document.querySelector('#entry-nav');
+var entriesNewButton = document.querySelector('#new-button');
+var entryFormContainer = document.querySelector('[data-view="entry-form"]');
+var entriesMainContainer = document.querySelector('[data-view="entries"]');
+
+entriesNewButton.addEventListener('click', entriesNewButtonHandler);
+
+function entriesNewButtonHandler(event) {
+  entryFormContainer.className = 'container';
+  entriesMainContainer.className = 'container hidden';
+}
+
+entriesNavItem.addEventListener('click', entriesNavItemHandler);
+
+function entriesNavItemHandler(event) {
+  entryFormContainer.className = 'container hidden';
+  entriesMainContainer.className = 'container';
+}
```

Comment:
Instead of making an individual function definition for each view, I would make one function named something like `switchViews` that will receive an argument of a `string` containing the name of the view to show. Within that function, you should loop over all the possible views and check if they are the view name given as an argument. If it is that view, you should show it, and if it isn't you should hide it. You would also want to assign a value to `data.view` for whichever view is the correct view. Once you've done this, you'll have a reusable function you can use any time you need to move between views, no matter the context.

Also, I see that you aren't making use of a `data-view` attribute at all in your code for the `a` elements that are used to move between views. I suggest going back to `javascript-view-swapping` to review how we handled this process there.

Reply:
Thanks Cody. Using your feedback, I was able to use the switchViews function to return to the page that the user was on even after a refresh.